airbyte-ci/connectors/pipelines/pipelines/airbyte_ci/connectors/test/steps/common.py

```diff
@@ -530,3 +533,64 @@ async def _build_regression_test_container(self, target_container_id: str) -> Co
 
         container = container.with_exec(["poetry", "lock", "--no-update"]).with_exec(["poetry", "install"])
         return container
+
+
+class RegressionTests(LiveTests):
+    """A step to run live tests for a connector."""
+
+    context: ConnectorContext
+    title = "Regression tests"
+
+    def test_command(self) -> List[str]:
+        """
+        This command:
+
+        1. Starts a Google Cloud SQL proxy running on localhost, which is used by the connection-retriever to connect to postgres.
+        2. Gets the PID of the proxy so it can be killed once done.
+        3. Runs the regression tests.
+        4. Kills the proxy, and waits for it to exit.
+        5. Exits with the regression tests' exit code.
+        We need to explicitly kill the proxy in order to allow the GitHub Action to exit.
+        An alternative that we can consider is to run the proxy as a separate service.
+
+        (See https://docs.dagger.io/manuals/developer/python/328492/services/ and https://cloud.google.com/sql/docs/postgres/sql-proxy#cloud-sql-auth-proxy-docker-image)
+        """
+        run_proxy = "./cloud-sql-proxy prod-ab-cloud-proj:us-west3:prod-pgsql-replica --credentials-file /tmp/credentials.json"
+        selected_streams = ["--stream", self.selected_streams] if self.selected_streams else []
+        run_pytest = " ".join(
+            [
+                "poetry",
+                "run",
+                "pytest",
+                "src/live_tests/regression_tests",
+                "--connector-image",
+                self.connector_image,
+                "--connection-id",
+                self.connection_id or "",
+                "--control-version",
+                self.control_version or "",
+                "--target-version",
+                self.target_version or "",
+                "--pr-url",
+                self.pr_url or "",
+                "--run-id",
+                self.run_id or "",
+                "--should-read-with-state",
+                str(self.should_read_with_state),
+                "--test-evaluation-mode",
+                str(self.test_evaluation_mode),
+            ]
+            + selected_streams
+        )
+        run_pytest_with_proxy = dedent(
+            f"""
+        {run_proxy} &
+        proxy_pid=$!
+        {run_pytest}
+        pytest_exit=$?
+        kill $proxy_pid
+        wait $proxy_pid
+        exit $pytest_exit
```

Comment:
📚 The explanation is a bit far from the weird stuff were doing. We should move this to another small function and likely discuss what were doing, and why, for future maintainers

Reply:
Good call-out. I've moved it into its own method and enhanced the docstring.